server2/app.py

```diff
@@ -0,0 +1,333 @@
+from functools import wraps
+
+from flask import Flask, request, redirect, url_for, make_response, Response
+
+import logging
+logging.basicConfig(level=logging.DEBUG)
+
+app = Flask(__name__)
+
+logger = app.logger
+
+
+algorithm = 'HS256'
+jwt_cookie_key = 'jwt_token'
+
+# not secret
+authorization_endpoint = os.environ.get('authorization_endpoint')
+token_endpoint = os.environ.get('token_endpoint')
+userinfo_endpoint = os.environ.get('userinfo_endpoint')
+redirect_uri = os.environ.get('redirect_uri')
+base_uri = os.environ.get('base_uri')
+default_object = os.environ.get('default_object')
+BUCKET_NAME = os.environ.get('bucket_name')
+
+# secrets
+if os.environ.get('DEBUG'):
+    env = 'dev'
+else:
+    env = 'prod'
+
+ssm_client = boto3.client('ssm')
+
+
+def get_secret_param(p):
+    return ssm_client.get_parameter(
+        Name='/{}/cloudefficiency/{}'.format(env, p),
+        WithDecryption=True
+    )["Parameter"]["Value"]
+
+
+client_id = get_secret_param('client_id')
+client_secret = get_secret_param('client_secret')
+jwt_secrets = get_secret_param('jwt_secrets').split(',')
+
+if len(client_id) == 0:
+    logger.error('Must pass a client_id stage variable')
+if len(client_secret) == 0:
+    logger.error('Must pass a client_secret stage variable')
+if len(jwt_secrets) == 0:
+    logger.error('Must pass a jwt_secrets stage variable')
+
+logger.debug('authorization_endpoint: %s', authorization_endpoint)
+logger.debug('token_endpoint: %s', token_endpoint)
+logger.debug('authorization_endpoint: %s', authorization_endpoint)
+logger.debug('userinfo_endpoint: %s', userinfo_endpoint)
+logger.debug('redirect_uri: %s', redirect_uri)
+logger.debug('base_uri: %s', base_uri)
+logger.debug('bucket_name: %s', BUCKET_NAME)
+
+oauth_token = "oauth_token"
+oauth_state = "oauth_state"
+oauth_token_key = base_uri + '/' + oauth_token
+oauth_state_key = base_uri + '/' + oauth_state
+oauth_initial_path_key = base_uri + '/initial_path'
+
+jwt_keys = [
+    "jti",
+    "iss",
+    "aud",
+    "exp",
+    "iat",
+    "nbf",
+    oauth_token_key,
+    oauth_state_key
+]
+
+
+def jwt_params(
+        oauth_token=None,
+        oauth_state=None,
+        initial_path=None,
+        sub=None,
+        iss=base_uri,
+        aud=base_uri,
+        exp_seconds=300,
+        **rest):
+    """
+    create a dict of jwt cliams
+    """
+    logger.debug('Creating jwt params')
+    now = datetime.utcnow()
+    expires = now + timedelta(seconds=exp_seconds)
+    token = rest
+    if initial_path:
+        token[oauth_initial_path_key] = initial_path
+    if oauth_token:
+        token[oauth_token_key] = oauth_token
+    if oauth_state:
+        token[oauth_state_key] = oauth_state
+    if sub:
+        token['sub'] = sub
+    token['jti'] = str(uuid.uuid1())
+    token['iss'] = base_uri
+    token['aud'] = base_uri
+    token['exp'] = expires
+    token['iat'] = now
+    token['nbf'] = now
+    logger.debug('JWT params: %s', token)
+    return token
+
+
+class Cookie:
```

Comment:
add a comment, explaining what the cookie class is doing. ideally add a key/value set for what the actually cookie would look like in say the browser view

Reply:
Done.